cupy/testing/helper.py

```diff
@@ -261,6 +514,46 @@ def for_int_dtypes(name='dtype', no_bool=False):
 
 
 def for_dtypes_combination(types, names=['dtype'], full=None):
+    """Decorator that checks the fixture with a product set of dtypes.
+
+    Args:
+         types(list of dtypes): dtypes to be tested.
+         names(list of str): Argument names to which dtypes are passed.
+         full(bool): If ``True``, then all combinations
+             of dtypes will be tested.
+             Otherwise, the subset of combinations will be tested
+             (see the description below).
+
+    Decorator adds the keyword arguments specified by ``names``
+    to the test fixture. Then, it runs the fixture with passing
+    (possibly, a subset of) a product set of dtypes.
+    The range of dtypes is specified by ``types``.
+
+    The combination of dtypes to be tested changes depending
+    on the option ``full``. If ``full`` is ``True``,
+    all combinations of ``types`` are tested.
+    Sometimes, such an exhaustive test can be costly.
+    So, if ``full`` is ``False``, only a subset of possible
+    combinations is tested. Specificially, at first,
+    the shuffled list of ``types`` are made for each argument.
+    Let the lists be ``D1``, ``D2``, ..., `Dn`
+    where :math:`n` is the number of arguments,
+    then, the combinations to be tested will be ``zip(D1, ..., Dn)``.
+    If ``full`` is ``None``, the behaivior is switched
+    by setting the environment variable ``CUPY_TEST_FULL_COMBINATION=1``.
+
+    For example, let ``types`` be ``[float16, float32, float64]``
+    and ``names`` be ``[a_type, b_type]``. If ``full`` is ``True``,
+    then the decorated test fixture is executed with all
+    :math:`2^3` patterns. On the other hand, if ``full`` is ``False``,
+    shuffled lists are made for ``a_type`` and ``b_type``.
+    Suppose the lists are ``(16, 64, 32)`` for ``a`` and
+    ``(32, 64, 16) for ``b`` (prefixes are removed for short).
+    Then the combination to be tested are
+    ``(a_type, b_type) = (16, 32), (64, 64),
```

Comment:
Like this?
 Then the combination for ``(a_type, b_type)`` to be tested are ``(16, 32)``, ``(64, 64)`` and ``(32, 16)``.